src/tokenizer.rs

```diff
@@ -101,40 +101,28 @@ where
 
 impl<'a, T> Tokens<'a, T>
 where
-    T: WktFloat + str::FromStr + Default,
+    T: str::FromStr,
 {
     fn read_until_whitespace(&mut self) -> Option<String> {
-        let mut result = String::new();
+        let mut result = None;
 
         while let Some(&next_char) = self.chars.peek() {
-            let marker = match next_char {
-                '\0' | '(' | ')' | ',' => true,
-                _ => false,
-            };
-
-            // Consume non-markers
-            if !marker {
-                let _ = self.chars.next();
-            }
-
-            let whitespace = is_whitespace(next_char);
-
-            // Append non-whitespace, non-marker characters
-            if !marker && !whitespace {
-                result.push(next_char);
-            }
-
-            // Stop reading when reached marker or whitespace
-            if marker || whitespace {
-                break;
+            match next_char {
+                '\0' | '(' | ')' | ',' => break, // Just stop on a marker
+                c if is_whitespace(c) => {
+                    let _ = self.chars.next();
+                    break;
+                }
+                _ => {
+                    let _ = self.chars.next();
+                    result
+                        .get_or_insert_with(|| String::with_capacity(16))
```

Comment:
Can you leave a code comment with the justification for this `16` constant?

Why do we need it at all? Why isn't it 8? or 32?

Reply:
Comment in next commit.